allensdk/brain_observatory/behavior/vbo_2023_nwb_writers/behavior/__main__.py

```diff
@@ -0,0 +1,144 @@
+    skip_stim: List[str],
+) -> dict:
+    """Load and write a BehaviorSession as NWB.
+
+    Check data object against associated data from the behavior_session
+    metadata table. Additionally, round trip the NWB file to confirm that it
+    was properly created for the session.
+
+    Parameters
+    ----------
+    behavior_session_id : int
+        Behavior session unique identifier.
+    behavior_session_metadata : dict
+        Dictionary of keys that overlap between the metadata in the
+        BehaviorSession object and the associated metadata table.
+    nwb_filepath : pathlib.Path
+        Base filename path to write the NWB to.
+    skip_metadata : list of str
+        List of metadata keys to skip when comparing data.
+    skip_stim : list of str
+        List of stimulus file keys to skip when comparing data.
+
+    Returns
+    -------
+    output_path : str
+        String path of where the NWB was written to.
+
+    Note
+    ----
+    Upon encountering an error, either from write or data comparison, the
+    code changes the file name suffix to error and does not return the
+    path of the file.
+    """
+
+    nwb_filepath_inprogress = str(nwb_filepath) + ".inprogress"
+    nwb_filepath_error = str(nwb_filepath) + ".error"
+
+    # Clean out files from previous runs:
+    for filename in [
+        nwb_filepath_inprogress,
+        nwb_filepath_error,
+        nwb_filepath,
+    ]:
+        if os.path.exists(filename):
+            os.remove(filename)
+
+    try:
+        lims_session = BehaviorSession.from_lims(behavior_session_id)
+        nwbfile = lims_session.to_nwb()
+        with NWBHDF5IO(nwb_filepath_inprogress, "w") as nwb_file_writer:
+            nwb_file_writer.write(nwbfile)
+
+        logging.info(
+            "Comparing a BehaviorSession created from LIMS "
+            "to data from the behavior session metadata and the "
+            "associated stimulus pickle file."
+        )
+        _compare_metadata(
+            input_session=lims_session,
+            table_metadata=behavior_session_metadata,
+            skip_metadata=skip_metadata,
+            skip_stim=skip_stim,
+        )
+
+        logging.info(
+            "Comparing a BehaviorSession created from LIMS "
+            "with a BehaviorSession created from NWB"
+        )
+        nwb_session = BehaviorSession.from_nwb_path(nwb_filepath_inprogress)
+        assert sessions_are_equal(lims_session, nwb_session, reraise=True)
+
+        os.rename(nwb_filepath_inprogress, nwb_filepath)
+        return {"output_path": str(nwb_filepath)}
+    except Exception as err:
+        if os.path.isfile(nwb_filepath_inprogress):
+            os.rename(nwb_filepath_inprogress, nwb_filepath_error)
+        raise err
+
+
+if __name__ == "__main__":
+
+    logging.basicConfig(
+        format="%(asctime)s - %(process)s - %(levelname)s - %(message)s"
+    )
+
+    args = sys.argv[1:]
+    try:
+        parser = argschema.ArgSchemaParser(
```

Comment:
FYI this is an old way of using Argschema. Usually we sublcass it like 

```
class NWBWriter(argschema.ArgsSchamaParser):
    default_schema = BehaviorInputSchema
    output_schema = OutputSchema

    def run(self):
        ....
```

and usually `input_data` is passed, not `args`